### DISCUSSION TITLE
Coulombic interactions and charges in HPMC

### DISCUSSION THREAD

**Discussion by PetersResearchGroup** [glotzerlab/hoomd-blue#1871] [General]
As best I can tell, coulombic interactions are not currently implemented for HPMC simulations. At the same time, the documentation for hoomd.hpmc.pair.Union includes "charges" in the body definition. Does this do anything? Are coulmbic interactions planned to be implemented for HPMC?

**Comment by joaander** [comment]
All particle-particle interactions in HPMC are restricted to `r < r_cut`. I have no plans to change this.

HPMC passes the individual particle charges from the system state (or the given constituent charges for `hpmc.pair.Union`) to the pair potential evaluation function. Currently, no [built-in HPMC potentials](https://hoomd-blue.readthedocs.io/en/v4.8.2/module-hpmc-pair.html) use charge. You can implement [a custom component](https://github.com/glotzerlab/hpmc-energy-template) that does (e.g. a cuttoff Yukawa potential with per-particle/constituent charges). You can also use the charge attribute to represent any per-particle parameter you would like in a custom potential.

**Reply by PetersResearchGroup** [reply]
Got it, that makes sense. Thank you.
